internal/services/keyvault/key_vault_certificates_data_source_test.go

```diff
@@ -0,0 +1,98 @@
+package keyvault_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+)
+
+type KeyVaultCertificatesDataSource struct{}
+
+func TestAccDataSourceKeyVaultCertificates_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "data.azurerm_key_vault_certificates", "test")
+	r := KeyVaultCertificatesDataSource{}
+
+	data.DataSourceTest(t, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).Key("names.#").HasValue("9"),
+			),
+		},
+	})
+}
+
+func (KeyVaultCertificatesDataSource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_key_vault_certificate" "test2" {
+  count = 30
+  name  = "certificate-${count.index}"
+  certificate_policy {
+    issuer_parameters {
+      name = "Self"
+    }
+
+    key_properties {
+      exportable = true
+      key_size   = 2048
+      key_type   = "RSA"
+      reuse_key  = true
+    }
+
+    lifetime_action {
+      action {
+        action_type = "AutoRenew"
+      }
+
+      trigger {
+        days_before_expiry = 30
+      }
+    }
+
+    secret_properties {
+      content_type = "application/x-pkcs12"
+    }
+
+    x509_certificate_properties {
+      key_usage = [
+        "cRLSign",
+        "dataEncipherment",
+        "digitalSignature",
+        "keyAgreement",
+        "keyEncipherment",
+        "keyCertSign",
+      ]
+
+      subject            = "CN=hello-world"
+      validity_in_months = 12
+    }
+  }
+  key_vault_id = azurerm_key_vault.test.id
+}
+
+resource "azurerm_key_vault_access_policy" "certificates" {
```

Comment:
There is an issue with this resource because we're managing certificate access policies inside of the [template](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/keyvault/key_vault_certificate_resource_test.go#L1315). If you remove `resource "azurerm_key_vault_access_policy" "certificates" {
` and add `"List"` to the linked block, this test should pass